File: src/BaseHmiTypes/Screens/Controls/HmiAlarmToolbarButton.cs
using BaseHmiTypes.Common;
using BaseHmiTypes.Screens.Base;

namespace BaseHmiTypes.Screens.Controls;

public sealed class HmiAlarmToolbarButton
{
    public HmiAlarmToolbarButtonType Type { get; set; }

    public string? SourceType { get; set; }

    public HmiProperty<bool>? Visible { get; set; }

    public HmiProperty<int>? Order { get; set; }

    public HmiMultilingualText? Caption { get; set; }

    public HmiMultilingualText? Tooltip { get; set; }

    public string? Format { get; set; }
}

File: typescript/base-hmi-types/src/screens/controls/HmiAlarmToolbarButton.ts
import { HmiMultilingualText } from "../../common/HmiMultilingualText.js";
import { HmiProperty } from "../base.js";
import { HmiAlarmToolbarButtonType } from "./HmiAlarmToolbarButtonType.js";

export class HmiAlarmToolbarButton {
  type = HmiAlarmToolbarButtonType.Unknown;
  sourceType?: string;
  visible?: HmiProperty<boolean>;
  order?: HmiProperty<number>;
  caption?: HmiMultilingualText;
  tooltip?: HmiMultilingualText;
  format?: string;
}
